nlp/classifier.js

```diff
@@ -0,0 +1,32 @@
+"use strict";
+
+const spawn = require('child_process').spawn;
+
+class NLPClassifier	{
+
+	constructor(){
+
+		this.pythonProcess = spawn('python3',['-u', "python_classifier/classifier.py"]);
+	}
+
+	async classify(input){
+
+		let process = this.pythonProcess;
+		var promise = new Promise((resolve, reject) => {
+			process.stdout.on('data', async (data)  =>  {
+
+				var receiveData = await data.toString();
```

Comment:
I'm afraid this approach is still problematic, and I would prefer you rework it.

First of all, every time you call `classify` you're assing a new listener to the stream, and never removing that listener. That listener uses memory, and this solution will grow in memory usage indefinitely, which is bad.

Additionally, you're relying on the data all coming in one chunk. This might true in tests, but there is no guarantee: the python process could return a bunch of things at once and then either the OS or node would split the buffer at arbitrary points. Calling toString on a buffer that was split at an arbitrary point can cause UTF8 decode errors, which is bad, and it can cause this function to return the wrong number of values (say 2 numbers instead of 4), which would cause all sort of errors down the line.

Even worse, concurrent requests are completely broken: if you start a second request before the first one has completed, you will use the data from the first response instead of the second. Even if you don't have multiple threads processing requests in python, you will have concurrent requests coming from the HTTP server side, so you need a serialization point.

Instead of `toString` on the buffer, call `setEncoding('utf8')` on the stream, so `data` will be already a string. Have a single `on('data')` handler and concatenate the buffer into a local string, then look for the record separator (`\n` in your case). Include a unique ID (e.g. a serial number) with every request and response, and use it match requests and responses.

If you do not want to implement the stream handling code yourself, please follow the example in https://github.com/stanford-oval/genie-toolkit/blob/master/lib/predictor.js and use `JsonDatagramSocket` (a copy is available in `util/json_datagram_socket.js`), which will let you read JSON objects directly, and not worry about some of these issues. You still need to have serial numbers to handle concurrent requests.

Also minor nit: data is a buffer, not a Promise of a string, so you don't need `await` here. Similarly, the signature of the event handler is a synchronous function, so you should not mark it `async`.